plugin/dns64/setup.go

```diff
@@ -0,0 +1,136 @@
+package dns64
+
+import (
+	"net"
+
+	"github.com/coredns/proxy"
+
+	"github.com/coredns/coredns/core/dnsserver"
+	"github.com/coredns/coredns/plugin"
+	clog "github.com/coredns/coredns/plugin/pkg/log"
+	"github.com/coredns/coredns/plugin/pkg/upstream"
+
+	"github.com/caddyserver/caddy"
+	"github.com/caddyserver/caddy/caddyfile"
+)
+
+var log = clog.NewWithPlugin("dns64")
+
+func init() {
+	caddy.RegisterPlugin("dns64", caddy.Plugin{
+		ServerType: "dns",
+		Action:     setup,
+	})
+}
+
+func setup(c *caddy.Controller) error {
+	dns64, err := dns64Parse(&c.Dispenser)
+	if err != nil {
+		return plugin.Error("dns64", err)
+	}
+
+	t := dnsserver.GetConfig(c).Handler("trace")
+	dns64.Trace = t
+	dnsserver.GetConfig(c).AddPlugin(func(next plugin.Handler) plugin.Handler {
+		dns64.Next = next
+		return dns64
+	})
+
+	for i := range *dns64.Upstreams {
+		u := (*dns64.Upstreams)[i]
+		c.OnStartup(func() error {
+			return u.Exchanger().OnStartup(dns64.Proxy)
+		})
+		c.OnShutdown(func() error {
+			return u.Exchanger().OnShutdown(dns64.Proxy)
+		})
+		// Register shutdown handlers.
+		c.OnShutdown(u.Stop)
+	}
+
+	return nil
+}
+
+func dns64Parse(c *caddyfile.Dispenser) (DNS64, error) {
+	_, defaultPref, _ := net.ParseCIDR("64:ff9b::/96")
+	dns64 := DNS64{
+		Proxy: &proxy.Proxy{
```

Comment:
this is done entirely different now; we just ask the coredns process to resolve it for us. Look into, uh, etcd or backend.go, pkg/upstream has the bits.